src/contract/table.ts

```diff
@@ -163,15 +176,17 @@ export class Table<TableRow extends ABISerializableConstructor = ABISerializable
      *  - `limit`: Maximum number of rows to return.
      * @returns {TableCursor<TableRow>} Promise resolving to a `TableCursor` of the table rows.
      */
-    first(limit: number): TableCursor<TableRow> {
+    first(limit: number, options: FindOptions = {}): TableCursor<TableRow> {
         const tableRowsParams = {
             table: this.name,
             limit,
             code: this.contract.account,
             type: this.rowType,
+            scope: options.scope,
         }
 
         return new TableCursor({
+            maxRows: limit,
```

Comment:
I find the use of maxRows a bit confusing (probably because it's used in the `first()` method, but not the `query()`. If maxRows will always be either equivalent to limit or undefined, why not just use the limit variable directly?

Reply:
The `first` call is the only place `maxRows` is set on the cursor (I believe?). 

That's because when you call `.first(10000).all()`, you'd expect to receive all 10000 rows. You're setting a max number of rows the cursor is going to iterate upon.

However, if you call `.first(10000).next()`, you should expect to only get back what the API is willing to return. It's not going to return all 10000 most likely, and I don't think we want `.next` to recursively call the API. `.next` in my mind is the "make one API call for this" or "load more" call for the cursor.